app/clients/sms/reach.py

```diff
@@ -0,0 +1,25 @@
+from app.clients.sms import SmsClient, SmsClientResponseException
+
+
+def get_reach_responses(status, detailed_status_code=None):
+    if status == 'TODO-d':
+        return ("delivered", "TODO: Delivered")
+    elif status == 'TODO-tf':
+        return ("temporary-failure", "TODO: Temporary failure")
+    elif status == 'TODO-pf':
+        return ("permanent-failure", "TODO: Permanent failure")
+    else:
+        raise KeyError
+
+
+class ReachClientResponseException(SmsClientResponseException):
+    pass  # TODO (custom exception for errors)
+
+
+class ReachClient(SmsClient):
```

Comment:
Should we add init_app and record_outcome to this boiler plate code?

Reply:
Eventually yes. The class isn't actually needed for this PR (callbacks), but I wanted to put something here in case this PR went in first.

My other PR about the sending side [does a bunch of refactoring to avoid copy/pasting init_app and record_outcome](https://github.com/alphagov/notifications-api/pull/3493/files#diff-95316b574f974237b3b7ce453fec09628bd1062087154789ff4d0176ea23c460R23).